plugin-build/src/main/kotlin/io/sentry/android/gradle/instrumentation/okhttp/visitor/OkHttpEventListenerMethodVisitor.kt

```diff
@@ -0,0 +1,63 @@
+package io.sentry.android.gradle.instrumentation.okhttp.visitor
+
+import io.sentry.android.gradle.instrumentation.MethodContext
+import org.objectweb.asm.MethodVisitor
+import org.objectweb.asm.Opcodes
+import org.objectweb.asm.commons.AdviceAdapter
+
+class OkHttpEventListenerMethodVisitor(
+    apiVersion: Int,
+    originalVisitor: MethodVisitor,
+    instrumentableContext: MethodContext
+) : AdviceAdapter(
+    apiVersion,
+    originalVisitor,
+    instrumentableContext.access,
+    instrumentableContext.name,
+    instrumentableContext.descriptor
+) {
+
+    override fun onMethodEnter() {
+        super.onMethodEnter()
+        // Add the following call at the beginning of the constructor with the Builder parameter:
+        // builder.eventListener(new SentryOkHttpEventListener(builder.eventListenerFactory));
+
+        // OkHttpClient.Builder is the parameter, retrieved here
+        visitVarInsn(Opcodes.ALOAD, 1)
+
+        // Let's declare the SentryOkHttpEventListener variable
+        visitTypeInsn(Opcodes.NEW, "io/sentry/android/okhttp/SentryOkHttpEventListener")
+
+        // Than we just copy it - I don't really know why it's needed, i just know it's needed
```

Comment:
the INVOKESPECIAL below will call the init (constructor) method of the `SentryOkHttpEventListener`; since it will consume the elements on the stack and won't push anything back on the stack (`<init>` returns void) dup is required to have a reference to the `SentryOkHttpEventListener` after the constructer has been called.

Reply:
thanks for the explanation! I'll add it to the comment 😄